test/test-onRequestFinished.js

```diff
@@ -0,0 +1,122 @@
+"use strict";
+
+const {deepEqual, equal, ok} = require("chai").assert;
+const sinon = require("sinon");
+
+const {setupTestDOMWindow} = require("./setup");
+
+describe("browser-polyfill", () => {
+  describe("wrapped devtools.network.onRequestFinished listener", () => {
+    it("does not wrap the listener if it is not a function", () => {
+      const fakeChrome = {
+        devtools: {
+          network: {
+            onRequestFinished: {
+              addListener: sinon.spy(),
+            },
+          },
+        },
+      };
+
+      return setupTestDOMWindow(fakeChrome).then(window => {
+        const fakeNonFunctionListener = {fake: "non function listener"};
+
+        const browserOnRequestFinished = window.browser.devtools.network.onRequestFinished;
+        browserOnRequestFinished.addListener(fakeNonFunctionListener);
+
+        const fakeChromeOnRequestFinished = fakeChrome.devtools.network.onRequestFinished;
+        deepEqual(
+          fakeChromeOnRequestFinished.addListener.firstCall.args[0],
+          fakeNonFunctionListener,
+          "The non-function listener has not been wrapped"
+        );
+      });
+    });
+
+    it("promisifies the result", () => {
+      const fakeChrome = {
+        devtools: {
+          network: {
+            onRequestFinished: {
+              addListener: sinon.spy(),
+              hasListener: sinon.stub(),
+              removeListener: sinon.spy(),
+            },
+          },
+        },
+      };
+
+      return setupTestDOMWindow(fakeChrome).then(window => {
+        const listener = sinon.spy();
+
+        const browserOnRequestFinished = window.browser.devtools.network.onRequestFinished;
+        browserOnRequestFinished.addListener(listener);
+
+        const fakeChromeOnRequestFinished = fakeChrome.devtools.network.onRequestFinished;
+        ok(fakeChromeOnRequestFinished.addListener.calledOnce,
+           "devtools.network.onRequestFinished.addListener should have been called once");
+
+        const wrappedListener = fakeChromeOnRequestFinished.addListener.firstCall.args[0];
+        wrappedListener({
+          getContent(cb) {
+            cb("<html>...</html>", "text/html; charset=utf8");
+          },
+        });
+
+        ok(listener.calledOnce, "listener has not been called");
```

Comment:
This and some other messages: make sure that the message describes the expected behavior.

`onRequestFinished has been called once` is a better description.

Reply:
Done.